### ISSUE TITLE
Utilize outside of Kodi?

### ISSUE THREAD

**Issue by bannsec** [eracknaphobia/plugin.video.psvue#7]
I'm trying to use this plugin outside of Kodi. Primarily, it seems you have done a fair amount of research into what calls need to be made, and how they all interact/what headers are needed/etc. I'm trying to determine how to utilize this to get a stream that I can send to my chromecast. There are other python libraries that handle sending the stream over to the cast, but i need to be able to figure out what the correct urls are to use.

**Comment by fivehorizons** [comment]
I’m looking for the same.   I saw your posts on livestream about this and wanted to see about you keep the cookie updated and the stream active.

**Comment by snoopyjoe** [comment]
Once the add-on makes the rights calls then the stream url I just a simple m3u8 file. I'm not sure how the Chromecast takes or sends info. Also I believe the Chromecast has a native psvue app.

**Comment by fivehorizons** [comment]
The links I have found the work for these sites have an m3u8 but they contain expiration times etc.

**Comment by rtirak** [comment]
@fivehorizons have you figuered anything out for this?

**Comment by mlburgett** [comment]
> 
> 
> Once the add-on makes the rights calls then the stream url I just a simple m3u8 file. I'm not sure how the Chromecast takes or sends info. Also I believe the Chromecast has a native psvue app.

@snoopyjoe is there a way to create a list of channels in that format, so they can be fed (for example) to a separate PVR backend as IPTV channels, instead of using IPTV Simple within Kodi?

**Comment by snoopyjoe** [comment]
You can get the m3u8 URL for each channel directly from the json on the psvue website without the add-on. Now even if you get all the urls in a playlist, there's no certainty the urls will play outside the device because you still need all the authorization like grant code, token code, cookie, npsso and device ID

**Comment by mlburgett** [comment]
Right, those would need to be provided in headers, when requesting the stream, I would imagine.  I was referring to  list with all the necessaries bagged up, so a proper request url could be formed.

**Comment by snoopyjoe** [comment]
With the authorization info as the headers? I don't think that's possible because the authorization is done in stages. Such as Passing a token to a URL to get a code, then using that code to authorize the device.

It might be possible with some special app out there but not to my knowledge
